### DISCUSSION TITLE
How do I pass Google idToken to the client?

### DISCUSSION THREAD

**Discussion by TaeyoonKwon** [nextauthjs/next-auth#2822] [Help] [answered]
### Question 💬
I am using an external express.js backend that I need to pass in idToken to get another jwt token from the backend.

However, I've found that there is no way I can get idToken...

Is there any way to accomplish this?

### How to reproduce ☕️

```
callbacks: {
    async signIn({ account }) {
      try {
        console.log(account.id_token) // How do I send it to the client-side?

        return true;
      } catch (e: any) {
        console.log(e?.response?.data || e);
        return false;
      }
    },
  }
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/configuration/callbacks#jwt-callback

https://next-auth.js.org/configuration/callbacks#session-callback

**Reply by TaeyoonKwon** [reply]
Thank you for the response. I've made it, but I feel the doc is little bit confusing.

It would make more sense if jwt creation is mentioned in client API doc where useSession is called.

**Reply by TaeyoonKwon** [reply]
When I first read it, I did not expect jwt callback to get called since I had no jwt configuration at all.

**Reply by balazsorban44** [reply]
Thanks for the feedback! JWT is the default way of persisting session. You could also use an adapter for database persistence.
